.github/workflows/publish-npm-packages.yml

```diff
@@ -0,0 +1,54 @@
+name: Publish npm packages
+
+on:
+    workflow_dispatch:
+        inputs:
+            wp_version:
+                description: 'WordPress major version'
+                required: true
+                type: string
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+    # The concurrency group contains the workflow name and the branch name for pull requests
+    # or the commit hash for any other events.
+    group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+    cancel-in-progress: true
+
+jobs:
+    wp-release:
+        name: WordPress major release
+        runs-on: ubuntu-latest
+        environment: WordPress packages
+        steps:
+            - name: Checkout (for CLI)
+              uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+              with:
+                  path: main
+                  ref: trunk
+
+            - name: Checkout (for publishing)
+              uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+              with:
+                  path: publish
+                  ref: wp/${{ github.event.inputs.wp_version }}
+                  token: ${{ secrets.GUTENBERG_TOKEN }}
+
+            - name: Configure git user name and email (for publishing)
+              run: |
+                  cd publish
+                  git config user.name "Gutenberg Repository Automation"
+                  git config user.email gutenberg@wordpress.org
+
+            - name: Setup Node
+              uses: actions/setup-node@5b52f097d36d4b0b2f94ed6de710023fbb8b2236 # v3.1.0
+              with:
+                  node-version-file: '.nvmrc'
+                  registry-url: 'https://registry.npmjs.org'
+
+            - name: Publish packages to npm ("wp/${{ github.event.inputs.wp_version }}" dist-tag)
+              run: |
+                  cd main
+                  npm ci
```

Comment:
For publishing we will need something like:

```
./bin/plugin/cli.js npm-wp --wp-version=wp-${{ github.event.inputs.wp_version }} --ci --repository-path ../publish
```

Reply:
For testing this could be something like
```
            - name: Publish build artifacts
              uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571 # v2.2.2
              if: always()
              with:
                  name: build
                  path: build
                  if-no-files-found: ignore
```

Reply:
^ this should allow us to download the build packages from GitHub in the same way as you can download screenshots of the failed e2e tests

Reply:
Thank you. I hope we won't need it. The job is nearly identical to the one we use for publishing after Gutenberg plugin RC1 release.